board.cpp

```diff
@@ -0,0 +1,164 @@
+    if(pos.xpos > board.width || pos.ypos > board.height) return false;
+    for(std::vector<Position>::iterator it = board.disabledPositions.begin(); it != board.disabledPositions.end(); it++){ //check if pos is not equal to an invalid position
+        if(*it == pos){
+            return false;
+        }
+    }
+    return true;
+}
+
+//gets all possible valid positions
+//mainly for bot logic
+std::vector<Position> getValidTiles(Board board){ 
+    std::vector<Position> validPositions;
+    for(int x = 1; x <= board.width; x++){
+        for(int y = 1; y <= board.height; y++){
+            Position pos;
+            pos.xpos = x;
+            pos.ypos = y;
+            if (isPositionValid(board, pos)){
+                validPositions.push_back(pos);
+            }
+        }
+    }
+    return validPositions;
+}
+
+//board construction
+Board createBoard(){
+    Board board;
+    bool created = false;
+    int width = NULL;
+    int height = NULL;
+    int mineCount = 0;
+    std::string pvx = "";
+    while (!(pvx == "pve") && !(pvx == "pvp")){
+        std::cout << "will this be a player vs player game or a player vs AI game?" << std::endl;
+        std::cout << "write 'pvp' for player vs player or 'pve' for player vs AI" << std::endl;
+        std::cin >> pvx;
+    }
+    while ((5 > width) || (width > 10)){
+        std::cout << "choose the width of the field (between 5 and 10)" << std::endl;
+        std::cin >> width;
+    }
+    while ((5 > height) || (height > 10)){
+        std::cout << "choose the height of the field (between 5 and 10)" << std::endl;
+        std::cin >> height;
+    }
+    while ((3 > mineCount) || (mineCount > 8)){
+        std::cout << "choose the number of mines on the field (between 3 and 8)" << std::endl;
+        std::cin >> mineCount;
+    }
+    board.width = width;
+    board.height = height;
+    board.playerCount = 2;
+    board.players = new Player[board.playerCount]; //dynamic array since we don't know the number of players beforehand
+    for(int p = 0; p < board.playerCount; p++){
+        Player player;
+        player.mineCount = mineCount;
+        player.id = p;
+        if(p==1 && pvx == "pve"){ //set player 2 to be controlled by AI, may need to be refactored when support for more than 2 players is implemented
+            player.isAI = true;
+        }
+        board.players[p] = player;
+        }
+    return board;
+}
+
+//print board to console
+//may be depreciated if an UI is implemented in the future
+void printField(Board board, int perspective){ //no se yo tampoco entiendo nada
+    bool isPositionEnabled;
+    for (int y = 0; y <= board.height; y++){ //para cada fila
+        if (y == 0){ //si es la primera fila, imprimimos solo espacio
+            std::cout << "   ";
+        }
+        else{ //caso contrario, imprimimos el numero de la fila
+            std::cout << y;
+            for(int space = 0; space < (3 - std::to_string(y).length()); space++){

```

Comment:
#should store the result of `3 - std::to_string(y).length()` into a variable, otherwise it'll be computed every loop